src/kOS/Communication/CommNetConnectivityManager.cs

```diff
@@ -105,15 +105,15 @@ public bool HasConnection(Vessel vessel1, Vessel vessel2)
                 return true;
 
             // We next need to query the network to find a connection between the two vessels.
-            // I found no exposed method for accessing chached paths directly, other than the
+            // I found no exposed method for accessing cached paths directly, other than the
             // control path.
 
             // WARNING: In stock this will only work for vessels with a relay antenna installed.
             // This is a limitation put in place to improve performance in the stock game, and
             // there isn't a very good way around it.
             var net = CommNetNetwork.Instance.CommNet;
             tempPath = new CommPath();
-            return net.FindPath(vessel1.Connection.Comm, tempPath, vessel2.Connection.Comm) || net.FindPath(vessel2.Connection.Comm, tempPath, vessel1.Connection.Comm);
+            return vessel1 == vessel2 || net.FindPath(vessel1.Connection.Comm, tempPath, vessel2.Connection.Comm) || net.FindPath(vessel2.Connection.Comm, tempPath, vessel1.Connection.Comm);
```

Comment:
Ah okay, another place where C#'s features are ones I do NOT consider a good idea.  The caller has to dig down deep into the implementation of a class to discover if  ``==`` is a reference equals or not.  It is, like 98% of the time, but to be sure about that other 2%, you have to explicitly call a method (``ReferenceEquals``) to do the more low-level thing of reference comparison.  That's massively un-intuitive (the thing that looks low level (``==``) is higher level than the thing that looks more high level (calling a method named something).